spring-context/src/main/java/org/springframework/scheduling/support/CronSequenceGenerator.java

```diff
@@ -149,8 +149,24 @@ public Date next(Date date) {
 		return calendar.getTime();
 	}
 
+	/**
+	 * Indicates whether the specified cron expression can be parsed into a
+	 * valid cron sequence generator
+	 * @param cronExpression the expression to evaluate
+	 * @return a boolean indicating whether the given expression is a valid cron
+	 *         expression
+	 */
+	public static boolean isValidExpression(String cronExpression) {
+		String[] fields = StringUtils.tokenizeToStringArray(cronExpression, " ");
+		return validateCronFields(fields);
+	}
+
+	private static boolean validateCronFields(String[] fields) {
+		return fields != null && fields.length == 6;
+	}
+
 	private void doNext(Calendar calendar, int dot) {
-		List<Integer> resets = new ArrayList<Integer>();
+		List<Integer> resets = new ArrayList<>();
```

Comment:
`master` is currently for Spring Framework 4.3 which supports Java 6, 7, and 8.

We therefore cannot use the `<>` operator.